app/vmstorage/servers/vmselect.go

```diff
@@ -89,7 +89,11 @@ func (api *vmstorageAPI) InitSearch(qt *querytracer.Tracer, sq *storage.SearchQu
 
 func (api *vmstorageAPI) SearchMetricNames(qt *querytracer.Tracer, sq *storage.SearchQuery, deadline uint64) ([]string, error) {
 	tr := sq.GetTimeRange()
-	maxMetrics := getMaxMetrics(sq.MaxMetrics)
+	maxMetrics := sq.MaxMetrics
+	// respect maxMetrics if it's already set
+	if maxMetrics <= 0 {
+		maxMetrics = getMaxMetrics(sq.MaxMetrics)
```

Comment:
What do you think about following changes in order to improve code clarity:
```suggestion
                 // fallback to maxUniqueTimeSeries if no limits were provided
                 // see https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7857
		maxMetrics = GetMaxUniqueTimeSeries()
```
In this way reader won't need to be involved in understanding of `getMaxMetrics` logic.